plugins/MauticCrmBundle/Integration/DynamicsIntegration.php

```diff
@@ -785,11 +792,13 @@ public function pushLeads($params = [])
             $totalUpdated += count($toUpdate);
             foreach ($toUpdate as $lead) {
                 if (isset($lead['email']) && !empty($lead['email'])) {
-                    $key                        = mb_strtolower($this->cleanPushData($lead['email']));
-                    $lead['integration_entity'] = $object;
-                    $leadsToUpdateInD[$key]     = $lead;
-                    $integrationEntity          = $this->em->getReference('MauticPluginBundle:IntegrationEntity', $lead['id']);
-                    $integrationEntities[]      = $integrationEntity->setLastSyncDate(new \DateTime());
+                    $key                                = mb_strtolower($this->cleanPushData($lead['email']));
+                    $lead['mauticContactTimelineLink']  = $this->getContactTimelineLink($lead['integration_entity_id']);
```

Comment:
This block of code is duplicated a lot throughout the plugins so let's move it to AbstractIntegration as a shared method.